components/script/dom/vr.rs

```diff
@@ -128,34 +131,112 @@ impl VR {
     }
 
     pub fn handle_webvr_event(&self, event: WebVREventMsg) {
-        let WebVREventMsg::DisplayEvent(event) = event;
-        match &event {
-            &webvr::VRDisplayEvent::Connect(ref display) => {
-                let display = self.sync_display(&display);
-                display.handle_webvr_event(&event);
-                self.notify_event(&display, &event);
-            },
-            &webvr::VRDisplayEvent::Disconnect(id) => {
-                if let Some(display) = self.find_display(id) {
-                    display.handle_webvr_event(&event);
-                    self.notify_event(&display, &event);
-                }
+        match event {
+            WebVREventMsg::DisplayEvent(event) => {
+                match &event {
+                    &WebVRDisplayEvent::Connect(ref display) => {
+                        let display = self.sync_display(&display);
+                        display.handle_webvr_event(&event);
+                        self.notify_event(&display, &event);
+                    },
+                    &WebVRDisplayEvent::Disconnect(id) => {
+                        if let Some(display) = self.find_display(id) {
+                            display.handle_webvr_event(&event);
+                        self.notify_event(&display, &event);
+                        }
+                    },
+                    &WebVRDisplayEvent::Activate(ref display, _) |
+                    &WebVRDisplayEvent::Deactivate(ref display, _) |
+                    &WebVRDisplayEvent::Blur(ref display) |
+                    &WebVRDisplayEvent::Focus(ref display) |
+                    &WebVRDisplayEvent::PresentChange(ref display, _) |
+                    &WebVRDisplayEvent::Change(ref display) => {
+                        let display = self.sync_display(&display);
+                        display.handle_webvr_event(&event);
+                    }
+                };
             },
-            &webvr::VRDisplayEvent::Activate(ref display, _) |
-            &webvr::VRDisplayEvent::Deactivate(ref display, _) |
-            &webvr::VRDisplayEvent::Blur(ref display) |
-            &webvr::VRDisplayEvent::Focus(ref display) |
-            &webvr::VRDisplayEvent::PresentChange(ref display, _) |
-            &webvr::VRDisplayEvent::Change(ref display) => {
-                let display = self.sync_display(&display);
-                display.handle_webvr_event(&event);
+            WebVREventMsg::GamepadEvent(event) => {
```

Comment:
Please introduce two new private methods `handle_display_event` and `handle_gamepad_event` and call them from here.